trin-execution/src/execution.rs

```diff
@@ -42,34 +42,38 @@ struct GenesisConfig {
 pub struct State {
     pub database: EvmDB,
     pub config: StateConfig,
-    next_block_number: u64,
+    execution_position: ExecutionPosition,
 }
 
-const GENESIS_STATE_FILE: &str = "resources/genesis/mainnet.json";
-
-impl Default for State {
-    fn default() -> Self {
-        Self::new(None, StateConfig::default())
-    }
-}
+const GENESIS_STATE_FILE: &str = "trin-execution/resources/genesis/mainnet.json";
+const TEST_GENESIS_STATE_FILE: &str = "resources/genesis/mainnet.json";
 
 impl State {
-    pub fn new(path: Option<PathBuf>, config: StateConfig) -> Self {
-        State {
-            next_block_number: 0,
-            config: config.clone(),
-            database: EvmDB::new(path, config).expect("Failed to create EVM database"),
-        }
+    pub fn new(path: Option<PathBuf>, config: StateConfig) -> anyhow::Result<Self> {
+        let (database, execution_position) =
+            EvmDB::new(path, config.clone()).expect("Failed to create EVM database");
```

Comment:
It's weird to have `EvmDB::new` return `execution_position` as well.
I would initialize / load `execution_position` here and pass it as an argument to `EvmDB::new`.

You will probably have to  do the same for the DB as well, but I would say it's better that way (why would EvmDB initialize db connection).